base/locks.jl

```diff
@@ -136,41 +184,45 @@ unsafe_convert(::Type{Ptr{Void}}, m::Mutex) = m.handle
 
 function _uv_hook_close(x::Mutex)
     h = x.handle
-    x.handle = C_NULL
-    ccall(:uv_mutex_destroy, Void, (Ptr{Void},), h)
-    Libc.free(h)
-    nothing
+    if h != C_NULL
```

Comment:
should some signal be indicated to the user if this is called when the handle is null?